File: lib/algorithms/selection_sort/selection_sort.rb
def selection_sort(arr)
  len = arr.length 
  return arr if len <= 1
  
  0.upto(len-1) do |i|
    minIndex = i
    i.upto(len-1) do |j|
      minIndex = j if arr[j] < arr[minIndex]
    end

    arr[i], arr[minIndex] = arr[minIndex], arr[i]
  end

  arr
end

p selection_sort([2, -2, 3, 50, 5])

File: lib/algorithms/selection_sort/selection_sort.py
def selection_sort(arr):
    length = len(arr)

    if length <= 1:
        return arr

    for i in range(length):
        min_index = i
        for j in range(i, length):
            if arr[j] < arr[min_index]:
                min_index = j
        arr[i], arr[min_index] = arr[min_index], arr[i]

    return arr


print(selection_sort([10, -2, 45, 3]))
